public/app/plugins/panel/timeline/TimelineChart.tsx

```diff
@@ -17,6 +25,17 @@ export interface TimelineProps extends Omit<GraphNGProps, 'prepConfig' | 'propsT
 
 const propsToDiff = ['mode', 'rowHeight', 'colWidth', 'showValue', 'alignValue'];
 
+const _preparePlotFrame = (frames: DataFrame[], dimFields: XYFieldMatchers) => {
+  frames.forEach((frame) => {
+    frame.fields.forEach((field) => {
+      field.config.custom = field.config.custom || {};
+      field.config.custom.spanNulls = -1;
+    });
+  });
```

Comment:
Can you explain why this is needed here, this should be handled by applyFieldOverrides already

Reply:
Ah, I think it's because spanNulls is not an official option for timeline ?

Reply:
this is because join/alignment can currently only expand nulls into `undefined` regions or remove them. both of these options are not good timeline because the nulls are explicit state terminations that should not removed or "grown" to fill additional space. so this is a bit of a magic constant for now to prevent nulls from being touched during joins. this will go away if/when the join step is removed.